iocage_lib/ioc_start.py

```diff
@@ -213,17 +214,15 @@ def __start_jail__(self):
             }, _callback=self.callback,
                 silent=self.silent)
 
-        if self.conf['vnet'] and self.defaultrouter == 'none':
-            self.log.debug('Grabbing default route')
-            self.defaultrouter = self.get_default_gateway()[0]
-            self.log.debug(f'Default Gateway: {self.defaultrouter}')
+        if self.conf['vnet'] and self.defaultrouter == 'auto':
+            self.log.debug('Grabbing IPv4 default route')
+            self.defaultrouter = self.get_default_ipv4_gateway()
```

Comment:
We should collapse `ipv4_gateway` and `ipv6_gateway` and instead supply a string to check against. Since they are very similar code blocks, with 6 requiring just a couple extra things.

Reply:
Do you mean merging
```
get_default_ipv4_gateway(self)
```
and
```
get_default_ipv6_gateway(self)
```
into one method with a signature like
```
get_default_ipv6_gateway(self, address_family=netifaces.AF_INET)
```
?

This was my first approach, but with respect to clean code patterns, I
decided to separate them:
1. Let the method name speak for itself instead of its probably
   obscure parameters.
2. Keep methods as small as possible and separate their concerns, even
   if they are similar.

But if you prefer a single method, I will implement it. In this case
it's a subtleness, anyway.

Reply:
Yeah, since I like the principle of DRY, I try to keep similar functions together. So I suggest a method called `get_default_gateway(self, address_family=netifaces.AF_INET)` and have it default like you did to ipv4, if the address family is 6, then do the 6 stuff as well.